### ISSUE TITLE
Icon software missing 

### ISSUE THREAD

**Issue by Pirriss** [yktoo/indicator-sound-switcher#90]
Hi,
has any one noticed the missing icon on the Desktop task bar?

**Comment by yktoo** [comment]
What is your OS, desktop environment and their versions?

**Comment by Pirriss** [comment]
OS: Ubuntu 18.04 bionic
Kernel: x86_64 Linux 4.15.0-106-generic
 Shell: bash 4.4.20
 DE: GNOME 
WM: GNOME Shell
 WM Theme: Adwait
GTK Theme: Ambiance [GTK2/3]

**Comment by yktoo** [comment]
Is it just the icon missing (blank) or no icon whatsoever?

**Comment by Pirriss** [comment]
The icon is missing but the button is there and clickable. Also the tray icon is not missing!

**Comment by Pirriss** [comment]
![Screenshot from 2020-06-17 09-29-38](https://user-images.githubusercontent.com/67017017/84924409-50ba2b00-b07d-11ea-876b-6f88949e4413.png)

**Comment by Pirriss** [comment]
![4](https://user-images.githubusercontent.com/67017017/84924831-f2da1300-b07d-11ea-9200-4e50b966f8a0.png)

**Comment by yktoo** [comment]
If I understood you correctly, the launcher icon is missing for dialogs but the tray icon is present?

Is the icon missing for both About and Preferences dialogs?

Is the file `/usr/share/icons/hicolor/scalable/apps/indicator-sound-switcher.svg` present?

**Comment by Pirriss** [comment]
Thank you for getting back to me,
Yes, the icon is there: _/usr/share/icons/hicolor/scalable/apps/indicator-sound-switcher.svg_
And yes missing launcher icon and tray icon present.

Preference:
![Screenshot from 2020-06-18 13-14-25](https://user-images.githubusercontent.com/67017017/85067582-c699c180-b165-11ea-8531-d1a7aa1f9c75.png)
Preference:
![Screenshot from 2020-06-18 13-14-37](https://user-images.githubusercontent.com/67017017/85067601-cd283900-b165-11ea-837a-9e13f4e492d1.png)
License:
![Screenshot from 2020-06-18 13-13-43](https://user-images.githubusercontent.com/67017017/85067625-d5807400-b165-11ea-895b-ad93c2d68bc1.png)

**Comment by rocha** [comment]
I have the same problem, but in my case the tray icon is also missing. The icon svg file is present and its permissions look correct.

**Comment by yktoo** [comment]
Hmm, cannot reproduce that. Here's a vanilla 18.04 install, both icons are present:
![image](https://user-images.githubusercontent.com/1675911/85522097-7e671d00-b605-11ea-90b0-2d0941adcf70.png)

Are you perhaps using a customised theme?

**Comment by Pirriss** [comment]
By the way, my Icon disappeared after new version I believe...

**Comment by rocha** [comment]
@yktoo is that an updated install? I think mine disappeared after a routine Ubuntu update. Granted, my setup is a bit convoluted (Stripped down Ubuntu Mate, running Qtile as WM).

**Comment by yktoo** [comment]
Yes, what's on the screenshot is the latest **2.3.4** release.

The icons have been cleaned up in 655479da, so you might try:
```bash
sudo cp /usr/share/icons/hicolor/scalable/apps/indicator-sound-switcher.svg /usr/share/icons/ubuntu-mono-dark/apps/22/
sudo cp /usr/share/icons/hicolor/scalable/apps/indicator-sound-switcher.svg /usr/share/icons/ubuntu-mono-light/apps/22/
sudo update-icon-caches /usr/share/icons/*
```

and see if it helps.

**Comment by rocha** [comment]
Thanks, that fixed the problem!

**Comment by Pirriss** [comment]
Great, it worked!
Thank you .
![1](https://user-images.githubusercontent.com/67017017/85746968-ba5cb980-b6bb-11ea-8a9a-6428c60adecc.png)             ![2](https://user-images.githubusercontent.com/67017017/85747030-c779a880-b6bb-11ea-9b76-ffc014babb5f.png)

**Comment by yktoo** [comment]
I'm happy that it worked but it doesn't make any sense to me. The `/usr/share/icons/hicolor` is the official default dir for icons, which is apparently not always landing well in Ubuntu.

Anyway, I'm currently not sure how and whether this is to be addressed structurally. Icons are implemented "by the book" in the SSI, so I'm closing this.

**Comment by rocha** [comment]
I only ran `sudo update-icon-caches /usr/share/icons/*` to make it work, so it looks like the culprit was the icon cache.

**Comment by yktoo** [comment]
After some investigation I think I understand what happened here.

The 2.3.1 release removed Ubuntu-specific dark and light icons as there was a symbolic one provided. Normally, `dh_icons` would add an invocation of `update-icon-caches` to the `postinst` and `postrm` maintainer scripts, but not for the "hicolor" ones, which is explicitly stated on `dh_icons`'s manpage:

> It takes care of adding maintainer script fragments to call update-icon-caches for icon directories. **(This is not done for gnome and hicolor icons, as those are handled by triggers.)**

So apparently, the obsolete icons were removed but that didn't trigger a cache update so Ubuntu would still try to find them at the former locations. And in versions **2.3.1** onward there's no update command anymore, as per the explanation above.

Long story short, should you have any issue with missing icons, just run:
```bash
sudo update-icon-caches /usr/share/icons/*
```

**Comment by rocha** [comment]
Thanks for taking the time to figure out what the problem was.

**Comment by paulodiovani** [comment]
This still happens to me, I have no icon (or black icon) on systray,
running `gtk4-update-icon-cache` didn't work.

Using Sway WM + Waybar.

List of existing icons:

```
-rw-r--r-- 1 root root 2529 Aug 29 16:29 /usr/share/icons/hicolor/128x128/apps/indicator-sound-switcher-symbolic.symbolic.png
-rw-r--r-- 1 root root  382 Aug 29 16:29 /usr/share/icons/hicolor/16x16/apps/indicator-sound-switcher-symbolic.symbolic.png
-rw-r--r-- 1 root root 5247 Aug 29 16:29 /usr/share/icons/hicolor/256x256/apps/indicator-sound-switcher-symbolic.symbolic.png
-rw-r--r-- 1 root root  713 Aug 29 16:29 /usr/share/icons/hicolor/32x32/apps/indicator-sound-switcher-symbolic.symbolic.png
-rw-r--r-- 1 root root 1028 Aug 29 16:29 /usr/share/icons/hicolor/48x48/apps/indicator-sound-switcher-symbolic.symbolic.png
-rw-r--r-- 1 root root 1331 Aug 29 16:29 /usr/share/icons/hicolor/64x64/apps/indicator-sound-switcher-symbolic.symbolic.png
-rw-r--r-- 1 root root 1698 Aug 29 16:29 /usr/share/icons/hicolor/scalable/apps/indicator-sound-switcher.svg
-rw-r--r-- 1 root root 1671 Aug 29 16:29 /usr/share/icons/hicolor/scalable/status/indicator-sound-switcher-symbolic.svg
-rw-r--r-- 1 root root 1671 Aug 29 16:29 /usr/share/icons/hicolor/symbolic/apps/indicator-sound-switcher-symbolic.svg
```

**Comment by inalone** [comment]
Exactly the same issue as the above comment, also on sway WM and Waybar
